### ISSUE TITLE
Test failure for data://test:test/,X in the data-url crate

### ISSUE THREAD

**Issue by djc** [servo/rust-url#626]
There's a panic here:

```
thread 'data: URL "data://test:test/,X"' panicked at 'Mime { type_: "text", subtype: "plain", parameters: [("charset", "US-ASCII")] }', .tests/wpt.rs:20:9
```

which appears to have been around at least since the data-url crate was moved into this repository around Feb 2018 (cc670f686b3cbaac0399383f788cd8ce032ec6cc).

**Comment by djc** [comment]
For some reason this does not show up as a test failure, which seems strange. Should investigate.

**Comment by djc** [comment]
Apparently this might be expected: https://github.com/servo/rust-url/pull/473.

**Comment by djc** [comment]
This was fixed by the changes in #473.
